### PR TITLE
feat: [cds-76860]: Support for fetching ServiceNow templates with searchkey, offset, pagenation

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Describe your changes
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52134)

### CODE REVIEW

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52134)* status: :shipit: complete! ✓

**Comment by vivekkarnatiharness** [comment]
trigger uts

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNgHelper.java [hunk 1]
```diff
@@ -445,10 +445,12 @@ private ServiceNowTaskNGResponse getTemplateList(ServiceNowTaskNGParameters serv
     ServiceNowConnectorDTO serviceNowConnectorDTO = serviceNowTaskNGParameters.getServiceNowConnectorDTO();
     ServiceNowRestClient serviceNowRestClient = getServiceNowRestClient(serviceNowConnectorDTO.getServiceNowUrl());
 
+    // Offset from serviceNowTaskNGParameters is equivalent to page number, So calculating the actual offset value.
+    int offSet = serviceNowTaskNGParameters.getTemplateListOffset() * serviceNowTaskNGParameters.getTemplateListLimit();
```
Comment (naman-goenka) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNgHelper.java:449]: Please do this in manager if possible

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNGParameters.java [hunk 2]
```diff
@@ -65,6 +65,7 @@ public class ServiceNowTaskNGParameters implements TaskParameters, ExecutionCapa
   List<String> delegateSelectors;
 
   String queryFields;
```
Comment (naman-goenka) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNGParameters.java:67]: nit: description

**Review by vivekkarnatiharness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52134)* status: 8 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNgHelper.java` line 449 at r2](https://reviewable.io/reviews/harness/harness-core/52134#gh-1309965619:-Nd4oB3q6LBCGua5Tcza:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/4e7b681c3a8fd20e58252c6d9256dbaea9eec7f6/930-delegate-tasks/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNgHelper.java#L449)):*
<details><summary><i>Previously, naman-goenka (Naman Goenka) wrote…</i></summary><blockquote>

Please do this in manager if possible
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNGParameters.java` line 67 at r2](https://reviewable.io/reviews/harness/harness-core/52134#gh-1309967872:-Nd4oCiRCaked9LCjQMH:b-w7niie) ([raw file](https://github.com/harness/harness-core/blob/4e7b681c3a8fd20e58252c6d9256dbaea9eec7f6/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/servicenow/ServiceNowTaskNGParameters.java#L67)):*
<details><summary><i>Previously, naman-goenka (Naman Goenka) wrote…</i></summary><blockquote>

nit: description
</blockquote></details>

Have added description to both fields

**Comment by vivekkarnatiharness** [comment]
trigger uts

**Review by abhinavmittal-harness — COMMENTED** [review]
> Reviewed 2 of 8 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52134)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained.

**Pull request merged**